libs/Matchmaking/test/Matchmaking.PeerDiscovery.Test/PeerDiscoveryTest.cs

```diff
@@ -290,18 +314,35 @@ public PeerDiscoveryTestReordered(ITestOutputHelper output) : base(null)
             uint SIO_UDP_CONNRESET = IOC_IN | IOC_VENDOR | 12;
             relay_.IOControl((int)SIO_UDP_CONNRESET, new byte[] { Convert.ToByte(false) }, null);
 
+            if (packetLoss)
+            {
+                dropCounters_ = new Dictionary<EndPoint, int>();
+            }
+
             Task.Run(async () =>
             {
                 try
                 {
                     while (true)
                     {
                         byte[] buf_ = new byte[1024];
-                        var result = await relay_.ReceiveAsync(new ArraySegment<byte>(buf_), SocketFlags.None);
+                        var result = await relay_.ReceiveFromAsync(new ArraySegment<byte>(buf_), SocketFlags.None, new IPEndPoint(IPAddress.Any, 0));
+
+                        if (dropCounters_ != null)
+                        {
+                            // Keep a counter for every endpoint. Forward on packet every `MaxRetries` from any endpoint and drop the rest.
```

Comment:
If I understand correctly, we're drop everything but the 'last try' packet. So we're more closely simulating a slow network than a lossy one? Is this because we need some to get through for the tests to pass? Would we get any more code coverage by introducing randomness? e.g. counter=0 -> 33% chance of send, counter=1 -> 66%, counter=2 -> 100%